pkg/proc/variables.go

```diff
@@ -1244,6 +1247,41 @@ func (v *Variable) maybeDereference() *Variable {
 	}
 }
 
+// loadPtr assumes that v is a pointer and loads its value. v also gets a child
+// variable, representing the pointed-to value. If v is already loaded,
+// loadPtr() is a no-op.
+func (v *Variable) loadPtr() {
+	if v.loaded {
+		return
+	}
+	v.loaded = true
+
+	t := v.RealType.(*godwarf.PtrType)
+	v.Len = 1
+
+	var child *Variable
+	if v.Unreadable == nil {
+		ptrval, err := readUintRaw(v.mem, v.Addr, t.ByteSize)
+		if err == nil {
+			child = v.newVariable("", ptrval, t.Type, DereferenceMemory(v.mem))
+		} else {
+			// We failed to read the pointer value; mark v as unreadable.
+			v.Unreadable = err
+			// Pointers get a child even if their value can't be read, to
+			// maintain backwards compatibility.
+			child = v.newVariable("", 0 /* addr */, t.Type, DereferenceMemory(v.mem))
+			child.Unreadable = fmt.Errorf("parent pointer unreadable: %w", err)
+		}
+	} else {
```

Comment:
Both this and the case above produce identical code, perhaps we could dry this up a bit. Maybe we should remove the `else` clause and then trigger the error condition on `v.Unreadable != nil` since that should get hit if it was already set or if the `readUintRaw` failed.

Reply:
done